File: tech-studio-projects/other-projects/document-management-system/java/src/main/java/com/dms/DocumentManager.java
package com.dms;

import java.util.ArrayList;
import java.util.List;

public class DocumentManager {
    private List<Document> documents;

    public DocumentManager() {
        this.documents = new ArrayList<>();
    }

    public void addDocument(Document document) {
        documents.add(document);
    }

    public Document getDocumentByTitle(String title) {
        for (Document doc : documents) {
            if (doc.getTitle().equalsIgnoreCase(title)) {
                return doc;
            }
        }
        return null;
    }
}

File: tech-studio-projects/other-projects/document-management-system/csharp/csharp/DocumentManager.cs
using System.Collections.Generic;
using System.Linq;

namespace DocumentManagementSystem
{
    public class DocumentManager
    {
        private readonly List<Document> documents;

        public DocumentManager()
        {
            documents = new List<Document>();
        }

        public void AddDocument(Document document)
        {
            documents.Add(document);
        }

        public Document GetDocumentByTitle(string title)
        {
            return documents.FirstOrDefault(doc => doc.Title.Equals(title, System.StringComparison.OrdinalIgnoreCase));
        }
    }
}
